src/data/index.ts

```diff
@@ -48,23 +48,100 @@ export type {
   QueryShared,
 } from './query';
 
+/**
+ * @typeParam T - The type of metadata associated with each record.
+ */
 export class Index<T extends RecordMetadata = RecordMetadata> {
   private config: PineconeConfiguration;
   private target: {
     index: string;
     namespace: string;
   };
 
+  /**
+   * Delete all records from the index.
+   *
+   * @example
+   * ```js
+   * const pinecone = new Pinecone();
+   * await pinecone.index('my-index').deleteAll();
+   * ```
+   * @returns A promise that resolves when the delete is completed.
+   */
   deleteAll: ReturnType<typeof deleteAll>;
+
+  /**
+   * Delete records from the index by either an array of ids, or a filter object.
+   * See [Filtering with metadata](https://docs.pinecone.io/docs/metadata-filtering#deleting-vectors-by-metadata-filter) for more on deleting records with filters.
+   *
+   * @example
+   * ```js
+   * const pinecone = new Pinecone();
+   * await pinecone.index('my-index').deleteMany(['record-1', 'record-2']);
+   *
+   * // or
+   * await pinecone.index('my-index').deleteMany({ filter: { genre: 'classical' }});
```

Comment:
```suggestion
   * await pinecone.index('my-index').deleteMany({ genre: 'classical' });
```

I'm pretty sure the `filter` key is not needed here. [It gets added before sending.](https://github.com/pinecone-io/pinecone-ts-client/blob/main/src/data/deleteMany.ts#L41)

Reply:
Good catch.